src/Pact/Eval.hs

```diff
@@ -591,16 +592,74 @@ moduleResolver lkp i mn = do
 
 
 resolveRef :: HasInfo i => i -> Name -> Eval e (Maybe Ref)
-resolveRef i (QName (QualifiedName q n _)) = moduleResolver (lookupQn n) i q
+resolveRef i (QName (QualifiedName q@(ModuleName refNs ns) n _)) = moduleResolver (lookupQn n) i q
   where
     lookupQn n' i' q' = do
       m <- lookupModule i' q'
-      return $ join $ HM.lookup n' . _mdRefMap <$> m
-resolveRef _i nn@Name {} = do
+      case m of
+        Nothing
+          | isJust ns -> evalError' i
+            $ "resolveRef: Unable to resolve module reference: "
+            <> pretty q
+          | otherwise -> do
+            let mn = ModuleName n (Just $ NamespaceName refNs)
+            md <- resolveModule i' mn
+            case md of
+              Just (ModuleData m'@MDModule{} _) -> do
+                let inf = getInfo i
```

Comment:
So ... we have `TModRef` now right? Let's go ahead and get rid of the `TModule` hacks. If you agree, there are other changes to this PR in how `TModule` is handled that can go away (e.g. in `typeOf`)